packages/xdl/src/ProjectAssets.ts

```diff
@@ -349,3 +395,29 @@ async function collectAssets(
 
   return [...bundles.ios.assets, ...bundles.android.assets, ...manifestAssets];
 }
+
+export async function resolveAndCollectExpoUpdatesManifestAssets(
+  projectRoot: string,
+  exp: ExpoConfig,
+  urlResolver: (path: string) => string
+): Promise<{ url: string; hash: string; key: string; contentType: string }[]> {
+  const manifestAssets: { url: string; hash: string; key: string; contentType: string }[] = [];
+  await resolveExpoUpdatesManifestAssets({
+    projectRoot,
+    manifest: exp,
+    async assetKeyResolver(assetPath) {
+      const absolutePath = path.resolve(projectRoot, assetPath);
+      const contents = await fs.readFile(absolutePath);
+      const fileType = await FileType.fromBuffer(contents);
+      const hash = md5hex(contents);
```

Comment:
I believe this is coupled to something (expo-asset?). Would be good to have a comment here to highlight that coupling since it's non-obvious why we use MD5 here.